server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java

```diff
@@ -175,6 +175,16 @@ public Entry(Entry entry, ImmutableOpenMap<ShardId, ShardSnapshotStatus> shards)
             this(entry, entry.state, shards, entry.failure);
         }
 
+        public Entry withRepoGen(long newRepoGen) {
+            return new Entry(snapshot, includeGlobalState, partial, state, indices, dataStreams, startTime, newRepoGen, shards, failure,
```

Comment:
Sure can, but it's not always `+1` (on master fail-over during finalization or delete we will skip a value of `N` here to avoid overwriting `index-N`)